### PR TITLE
Added unit tests for retrieveProducts and purchase

### PR DESCRIPTION
**Pull request by cb-amutha**
Unit tests:
- retrieveProducts
- purchaseProducts
- done flutter formattings for the unit test class
Github Actions:

 - unit test added on PR 
Entitlement API:

 - return type updated list of CBEntitlementWrapper model instead of String

### CODE REVIEW

File: .github/workflows/ci-checks.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          java-version: 11
+      - name: Setup Flutter Environment
+        uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable'
+          flutter-version: '3.3.10'
+      - name: Get Flutter Dependencies
+        run: flutter pub get
+      - name: Unit Test
+        run: flutter test ${GITHUB_WORKSPACE}/test/chargebee_test.dart
```
Comment (cb-haripriyan) [inline .github/workflows/ci-checks.yml:29]: Can we make this generic to include all files inside the test folder?

Comment (cb-amutha) [reply .github/workflows/ci-checks.yml]: Yes, Done!

File: .github/workflows/ci-checks.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
```
Comment (cb-haripriyan) [inline .github/workflows/ci-checks.yml:16]: Do we have any dependency on v1? if not can we use the latest ones?

File: lib/src/chargebee.dart [hunk 3]
```diff
@@ -155,11 +155,11 @@ class Chargebee {
...
   /// The list of entitlement details be returned if api success.
   /// Throws an [PlatformException] in case of failure.
-  static Future<List<String>> retrieveEntitlements(
+  static Future<List<CBEntitlementWrapper>> retrieveEntitlements(
       Map<String, String> queryParams,) async {
     final String result =
         await platform.invokeMethod(Constants.mGetEntitlements, queryParams);
-    return CBEntitlementWrapper.fromJson(jsonDecode(result)).entitlementsList;
+    return CBEntitlementList.fromJson(jsonDecode(result)).entitlementsList;
   }
```
Comment (cb-haripriyan) [inline lib/src/chargebee.dart:163]: Can we update the example/readme docs accordingly?

File: .github/workflows/ci-checks.yml (hunk 2 above)
Comment (cb-amutha) [reply .github/workflows/ci-checks.yml:16]: I think we don't have any dependency. lets check and update v3

File: lib/src/models/product.dart [hunk 4]
```diff
@@ -167,11 +167,11 @@ class CBProductIdentifierWrapper {
...
-  late String featureId;
-  late String featureName;
-  late String featureType;
-  late String value;
-  late String name;
+  late String? featureId;
+  late String? featureName;
+  late String? featureType;
+  late String? value;
+  late String? name;
```
Comment (cb-haripriyan) [inline lib/src/models/product.dart:174]: Are all of these optional? The `featureId`, `featureName` and `featureType` are not optional AFAIK. Lets check and update it accordingly.

**Pull request merged**